### PR TITLE
Add support to variable with [] inside type

### PR DESCRIPTION
**Pull request by bansan85**
Like `std::unique_ptr<int[]> variable;`

### CODE REVIEW

**Comment by virtuald** [comment]
Needs tests. 

Also, have you tried https://github.com/robotpy/cxxheaderparser? It supports constructs like that already.

**Comment by bansan85** [comment]
Sure. I will update the PR.

I use this deprecated library because the project https://github.com/thibaultmarin/hpp2plantuml uses it.

I will (try to) send another PR to add support about `int i {0};`. The field is currently skipped.

**Comment by bansan85** [comment]
I also fixed two tests.

I just have a doubt. The output format is `std::unique_ptr<int [ ] >`. Is it fine ?

**Comment by bansan85** [comment]
I used the same MR to add support for initialization with {}. Hope it's fine.  Review by commit will be easier than review the whole MR at once.

File: test/test_CppHeaderParser.py [hunk 1]
```diff
@@ -1162,14 +1162,9 @@ def test_method_params(self):
             [],
         )
 
-    def test_class_template(self):
-        self.assertEqual(
```
Comment (virtuald) [inline test/test_CppHeaderParser.py:1166]: Hm? I don't really want to break existing users without a good reason. Why are existing tests changing?

**Comment by virtuald** [comment]
Also, please merge in main, which fixes github actions.

File: test/test_CppHeaderParser.py (hunk 1 above)
Comment (bansan85) [reply test/test_CppHeaderParser.py:1166]: I think it's a side effect of the duplicated class `CarrotClass_TestCase`. When the class was duplicated, function `test_class_template` throws an exception but it was ignored by Python. When I renamed the second class to `ExternClass_TestCase`, Python catchs the exception and test fails.

If you look at class `CarrotClass` in header file you can see that the class is not template.

**Comment by bansan85** [comment]
> Also, please merge in main, which fixes github actions.

I merge my branch `square_bracket_in_type` in my `main` branch. But I'm not really sure it's what you want...

**Comment by virtuald** [comment]
Looks like you broke a test.

**Pull request merged**